### PR TITLE
Optimize type ID generation, remove location ID

### PR DESCRIPTION
**Pull request by turbolent**
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: runtime/common/addresslocation.go [hunk 1]
```diff
@@ -80,12 +80,30 @@ func (l AddressLocation) MeteredID(memoryGauge MemoryGauge) LocationID {
...
 func (l AddressLocation) TypeID(memoryGauge MemoryGauge, qualifiedIdentifier string) TypeID {
-	return NewMeteredTypeID(
-		memoryGauge,
-		AddressLocationPrefix,
-		l.Address.Hex(),
-		qualifiedIdentifier,
-	)
+	var i int
+
+	// AddressLocationPrefix '.' hex-encoded address '.' qualifiedIdentifier
```
Comment (dreamsmasher) [inline runtime/common/addresslocation.go:85]: the TypeID impls for AddressLocation, ScriptLocation, and TransactionLocation seem almost identical save for the prefices and middle fields

Comment (dsainati1) [reply runtime/common/addresslocation.go:85]: Agree it would be nice to refactor these to use a common helper if possible.

Comment (turbolent) [reply runtime/common/addresslocation.go]: Good idea, refactored in 36afaf932b06df966f5ac43d8429c6fee4ce1172

**Review by dsainati1 — APPROVED** [review]

File: runtime/common/address.go [hunk 2]
```diff
@@ -61,6 +61,12 @@ func (a Address) Hex() string {
 	return fmt.Sprintf("%x", a[:])
 }
 
+// HexInto encodes the address as a hex string representation of the address
+// into the given target byte slice.
+func (a Address) HexInto(target []byte) {
```
Comment (robert-e-davidson3) [inline runtime/common/address.go:66]: This naming is less confusing for me:

```suggestion
// ToHex encodes the address as a hex string representation of the address
// into the given target byte slice.
func (a Address) ToHex(target []byte) {
```

**Review by robert-e-davidson3 — APPROVED** [review]

File: runtime/common/addresslocation.go [hunk 3]
```diff
@@ -80,10 +80,11 @@ func (l AddressLocation) MeteredID(memoryGauge MemoryGauge) LocationID {
 }
 
 func (l AddressLocation) TypeID(memoryGauge MemoryGauge, qualifiedIdentifier string) TypeID {
-	return NewMeteredTypeID(
+	return idLocationTypeID(
```
Comment (SupunS) [inline runtime/common/addresslocation.go:83]: Can the old method be replaced with the new one?

Is the new method, and the old one essentially produce the same output? If the old method will not be removed, then maybe add a test case to ensure both produce identical results

Comment (turbolent) [reply runtime/common/addresslocation.go]: It couldn't be removed yet, as REPL, string, and identifier locations were still relying on it. 
I optimized those as well in cbbad360e4d20444fe67403f5bad41f25a4bfcd5, and also removed the unnecessary location ID code along the way.

File: runtime/common/address.go (hunk 2 above)
Comment (turbolent) [reply runtime/common/address.go]: This wasn't actually needed, so I removed it

**Pull request merged**